extensions/eclipse-che-theia-remote/src/node/plugin-remote.ts

```diff
@@ -0,0 +1,225 @@
+/*********************************************************************
+ * Copyright (c) 2018-2019 Red Hat, Inc.
+ *
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ **********************************************************************/
+
+import * as http from 'http';
+import * as ws from 'ws';
+import 'reflect-metadata';
+import { logger } from '@theia/core';
+import { ILogger } from '@theia/core/lib/common';
+import { Emitter } from '@theia/core/lib/common/event';
+import { MAIN_RPC_CONTEXT, PluginDeployer, PluginDeployerEntry, PluginMetadata } from '@theia/plugin-ext';
+import pluginVscodeBackendModule from '@theia/plugin-ext-vscode/lib/node/plugin-vscode-backend-module';
+import { RPCProtocolImpl } from '@theia/plugin-ext/lib/api/rpc-protocol';
+import { PluginDeployerHandler } from '@theia/plugin-ext/lib/common';
+import { PluginHostRPC } from '@theia/plugin-ext/lib/hosted/node/plugin-host-rpc';
+import { HostedPluginReader } from '@theia/plugin-ext/lib/hosted/node/plugin-reader';
+import pluginExtBackendModule from '@theia/plugin-ext/lib/plugin-ext-backend-module';
+import { Container, inject, injectable } from 'inversify';
+import { DummyTraceLogger } from './dummy-trace-logger';
+import { HostedPluginRemote } from './hosted-plugin-remote';
+import pluginRemoteBackendModule from './plugin-remote-backend-module';
+import { TerminalContainerAware } from './terminal-container-aware';
+
+/**
+ * Entry point of a Remote Endpoint. It is executed as a new separate nodejs process.
+ * It is using inversify to bind all the stuff.
+ * @author Florent Benoit
+ */
+
+process.on('SIGINT', () => {
+    process.exit();
+});
+
+// configured port number
+const PLUGIN_PORT = parseInt(process.env.THEIA_PLUGIN_ENDPOINT_PORT || '2503', 10);
+
+// start websocket server
+const WebSocketServerImpl = ws.Server;
+
+// display message about process being started
+console.log(`Theia Endpoint ${process.pid}/pid listening on port`, PLUGIN_PORT);
+const webSocketServer = new WebSocketServerImpl({ port: PLUGIN_PORT });
+
+interface CheckAliveWS extends ws {
```

Comment:
Please group declaration and commands.
Or even put all the entrypoint commands into separate function.

Reply:
ok will check. For info it was like that already in the other repository.

Reply:
moved console.log command at the end of the file